crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+use move_core_types::language_storage::StructTag;
+use move_core_types::value::{self, MoveStruct, MoveTypeLayout};
+use sui_indexer::models_v2::objects::StoredObject;
+use sui_types::dynamic_field::DynamicFieldInfo;
+use sui_types::{dynamic_field::DynamicFieldType, TypeTag};
+
+use super::{
+    base64::Base64, move_object::MoveObject, move_value::MoveValue, sui_address::SuiAddress,
+};
+use crate::context_data::package_cache::PackageCache;
+use crate::error::{code, Error};
+use crate::{context_data::db_data_provider::PgManager, error::graphql_error};
+use sui_types::object::Object as NativeSuiObject;
+
+pub(crate) struct DynamicField {
+    pub stored_object: StoredObject,
+}
```

Comment:
In the PR where I'm changing the storage representation of the other `Object`-based types, I adopt a pattern where Object "sub"-types, store the full contents of "Object", (like you are doing here), but then also the fields that are unique to them, in a deserialized/structured form, in this case the deserialized forms of some of the `df_*` columns:

```rust
pub(crate) struct  DynamicField {
    pub stored: StoredObject,
    pub df_kind: DynamicFieldType,
    pub df_object_id: SuiAddress,
}
```

I would leave out `df_name` and `df_object_type` because they could be arbitrarily big and you don't need to do much to them to convert them (beyond wrapping them in the appropriate `GraphQL` type), but proactively deserializing the other fields will help avoid your code being scattered with checks that they exist -- you do that once, when creating the `DynamicField` and then never again, and you can rely on the fact that they exist to offer APIs to expose the name and object type that can always treat the failure to access them as an internal error (because no row should contain a df_kind but not a df_name, etc).

Reply:
Word, in my initial approach I had adopted a similar pattern ... not sure why I dropped it. Agree that this would be easier than the crazy number of conditional checks in this current PR